### PR TITLE
#165981128 Refactor Server side code

### PR DESCRIPTION
**Pull request by Jonathan4github**
- restructure routes
- use map instead of filter where appropriate
- destructured and rename objects
- rename find method to decriptive and singular noun
- implement appropriate message for returns
- fix loan id validation

[Finishes #165981128]

### CODE REVIEW

File: server/controllers/LoanController.js [hunk 1]
```diff
@@ -14,37 +14,35 @@ class LoanController {
     const loan = loanData.find(loans => loans.id === parseInt(req.params.id, 10));
     return res.status(200).json({
       status: 200,
+      message: 'Retrieve successfull',
```
Comment (frostyblok) [inline server/controllers/LoanController.js]: Fix typo

File: server/controllers/LoanController.js [hunk 2]
```diff
@@ -93,6 +92,7 @@ class LoanController {
     loan.status = status;
     return res.status(200).json({
       status: 200,
+      message: 'Loan status was updated successful',
```
Comment (frostyblok) [inline server/controllers/LoanController.js]: "Loan updated successfully"

**Review by frostyblok — COMMENTED** [review]
Good job, @Jonathan4github. Please attend to the minor comments.

File: server/controllers/LoanController.js (hunk 1 above)
Comment (Jonathan4github) [reply server/controllers/LoanController.js]: Noted

File: server/controllers/LoanController.js (hunk 2 above)
Comment (Jonathan4github) [reply server/controllers/LoanController.js]: Noted

**Pull request merged**